cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -154,266 +145,16 @@ def finish_all_flow_cells_temp(self) -> bool:
-            lane=lane,
-        )
-
-        if metric:
-            return metric.sample_base_fraction_passing_q30 >= q30_threshold / 100
-
-        LOG.warning(
-            f"Skipping fastq file {sample_fastq_path.name} as no metrics entry was found in status db."
-        )
-        LOG.warning(f"Flow cell name: {flow_cell_name}, sample id: {sample_id}, lane: {lane} ")
-        return False
-
-    def add_sample_sheet_path_to_housekeeper(
-        self, flow_cell_directory: Path, flow_cell_name: str
-    ) -> None:
-        """Add sample sheet path to Housekeeper."""
-
-        try:
-            sample_sheet_file_path: Path = get_sample_sheet_path(
-                flow_cell_directory=flow_cell_directory
-            )
-
-            self.add_file_to_bundle_if_non_existent(
-                file_path=sample_sheet_file_path,
-                bundle_name=flow_cell_name,
-                tag_names=[SequencingFileTag.SAMPLE_SHEET, flow_cell_name],
-            )
-        except FileNotFoundError as e:
-            LOG.error(
-                f"Sample sheet for flow cell {flow_cell_name} in {flow_cell_directory} was not found, error: {e}"
-            )
-
-    def add_bundle_and_version_if_non_existent(self, bundle_name: str) -> None:
-        """Add bundle if it does not exist."""
-        if not self.hk_api.bundle(name=bundle_name):
-            self.hk_api.create_new_bundle_and_version(name=bundle_name)
-
-    def add_tags_if_non_existent(self, tag_names: List[str]) -> None:
-        """Ensure that tags exist in Housekeeper."""
-        for tag_name in tag_names:
-            if self.hk_api.get_tag(name=tag_name) is None:
-                self.hk_api.add_tag(name=tag_name)
-
-    def add_file_to_bundle_if_non_existent(
-        self, file_path: Path, bundle_name: str, tag_names: List[str]
-    ) -> None:
-        """Add file to Housekeeper if it has not already been added."""
-        if not file_path.exists():
-            LOG.warning(f"File does not exist: {file_path}")
-            return
-
-        if not self.file_exists_in_latest_version_for_bundle(
-            file_path=file_path, bundle_name=bundle_name
-        ):
-            self.hk_api.add_and_include_file_to_latest_version(
-                bundle_name=bundle_name,
-                file=file_path,
-                tags=tag_names,
-            )
-
-    def file_exists_in_latest_version_for_bundle(self, file_path: Path, bundle_name: str) -> bool:
-        """Check if file exists in latest version for bundle."""
-        latest_version: Version = self.hk_api.get_latest_bundle_version(bundle_name=bundle_name)
-        return any(
-            file_path.name == Path(bundle_file.path).name for bundle_file in latest_version.files
+        store_flow_cell_data_in_status_db(parsed_flow_cell, store=self.status_db)
+        store_sequencing_metrics_in_status_db(parsed_flow_cell, store=self.status_db)
+        update_sample_read_counts_in_status_db(parsed_flow_cell, store=self.status_db)
```

Comment:
We are using a mix of positional and key word arguments here. I'm fine with it since the combination of variable and function names makes it clear what's happening, but it does go against the discussion we had in the coding guidelines doc. What do you think @ChrOertlin and @seallard?

Reply:
You are absolutely right, I fixed this!